dot/sync/outliers.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"math/big"
+	"sort"
+)
+
+// removeOutlier removes the outlier from the slice
+// Explanation:
+// IQR outlier detection
+// Q25 = 25th_percentile
+// Q75 = 75th_percentile
+// IQR = Q75 - Q25         // inter-quartile range
+// If x >  Q75  + 1.5 * IQR or  x   < Q25 - 1.5 * IQR THEN  x is a mild outlier
+// If x >  Q75  + 3.0 * IQR or  x   < Q25 – 3.0 * IQR THEN  x is a extreme outlier
+// Ref: http://www.mathwords.com/o/outlier.htm
+//
+// returns: sum of all the non-outliers elements
+func removeOutlier(dataArr []*big.Int) (sum *big.Int, count int64) {
+	length := len(dataArr)
+
+	switch length {
+	case 0:
+		return big.NewInt(0), 0
+	case 1:
+		return dataArr[0], 1
+	case 2:
+		return big.NewInt(0).Add(dataArr[0], dataArr[1]), 2
+	}
+
+	sort.Slice(dataArr, func(i, j int) bool {
+		return dataArr[i].Cmp(dataArr[j]) < 0
+	})
+
+	half := length / 2
+	data1 := dataArr[:half]
```

Comment:
could you rename `data1` to `firstHalf` and `data2` to `secondHalf`? I belive this can improve the readability!

Reply:
done ✔